File: Cc.Ui/src/models/eventType.ts
import { Entity, resource, field, required } from '../framework/entity';
import { Pagination, Pager } from '../framework/core/actions';

@resource('eventType')
export class EventType extends Entity {

    @field()
    id: number;
    @field()
    name: string;
    @field()
    description: string;
    @field()
    insertUser: string;
    @field()
    insertDate: string;
    @field()
    updateUser: string;
    @field()
    updateDate: string;

    constructor(data?: any) {
        super()
        this.init(data);      
    }

    static newItem(): EventType {

        let item = new EventType();
        return item;
    }
}

@resource('eventType')
export class EventTypeSearchModel extends Entity implements Pagination {
    
    @field()
    name?: string;
    @field()
    description?: string;

    @field()
    pager: Pager = <Pager> {
        take: 100,
        orderBy: 'id'
    };
}

File: Cc.Core/Models/EventType.cs
using System;
using System.Collections.Generic;

namespace Cc.Core.Models
{
    public partial class EventType
    {
        public EventType()
        {
            Event = new HashSet<Event>();
        }

        public long Id { get; set; }
        public string Name { get; set; } = null!;
        public string? Description { get; set; }
        public string? InsertUser { get; set; }
        public DateTime? InsertDate { get; set; }
        public string? UpdateUser { get; set; }
        public DateTime? UpdateDate { get; set; }

        public virtual ICollection<Event> Event { get; set; }
    }
}
